MatrixKit/Controllers/MXKAttachmentsViewController.m

```diff
@@ -465,6 +499,17 @@ - (void)refreshCurrentVisibleCell
     
     [self refreshCurrentVisibleItemIndex];
     
+    // Tell the delegate that a new Attachment has been shown and pass eventId
+    MXKMediaCollectionViewCell* cell = [[self.attachmentsCollection visibleCells] firstObject];
```

Comment:
@aramsargsyan: Instead of using the `visibleCells` array here, I would prefer to handle the delegate call during the currentVisibleItemIndex checking. Here is my suggested code for the`refreshCurrentVisibleCell` method:
```
- (void)refreshCurrentVisibleCell
{
    // In case of attached image, load here the high res image.
    
    [self refreshCurrentVisibleItemIndex];
    
    if (currentVisibleItemIndex == NSNotFound)
    {
        // Tell the delegate that a no attachment is displayed for the moment.
        if ([self.delegate respondsToSelector:@selector(displayedNewAttachmentWithEventId:)])
        {
            [self.delegate displayedNewAttachmentWithEventId:nil];
        }
    }
    else
    {
        NSInteger item = currentVisibleItemIndex;
        if (isBackPaginationInProgress)
        {
            if (item == 0)
            {
                // Tell the delegate that a no attachment is displayed for the moment.
                if ([self.delegate respondsToSelector:@selector(displayedNewAttachmentWithEventId:)])
                {
                    [self.delegate displayedNewAttachmentWithEventId:nil];
                }
				
                return;
            }
            
            item --;
        }
        
        if (item < attachments.count)
        {
            MXKAttachment *attachment = attachments[item];
            NSString *attachmentURL = attachment.actualURL;
            NSString *mimeType = attachment.contentInfo[@"mimetype"];
			
            // Tell the delegate which attachment is currently displayed by using its event id.
            if ([self.delegate respondsToSelector:@selector(displayedNewAttachmentWithEventId:)])
            {
                [self.delegate displayedNewAttachmentWithEventId:attachment.eventId];
            }
            
            // Check attachment type
            if (attachment.type == MXKAttachmentTypeImage && attachmentURL.length && ![mimeType isEqualToString:@"image/gif"])
            {
                [...]
            }
        }
    }
}
```